### PR TITLE
fix:can't switch sync mode when no snap-sync peer

### PR DESCRIPTION
**Pull request by lochjin**

### CODE REVIEW

File: services/common/flags.go [hunk 1]
```diff
@@ -686,6 +691,7 @@ func DefaultConfig(homeDir string) *config.Config {
 		SubmitNoSynced:       false,
 		DevNextGDB:           true,
 		GBTTimeOut:           defaultGBTTimeout,
+		SnapTimeout:          300,
```
Comment (dindinw) [inline services/common/flags.go]: Better to add extra `defaultSnapTimeout`  after line 43 : `defaultGBTTimeout = 800`

File: config/config.go [hunk 2]
```diff
@@ -144,6 +144,8 @@ type Config struct {
 
 	// TODO: It will soon be discarded in the near future
 	DevSnapSync bool `long:"devsnapsync" description:"Enable snap sync for P2P that only exist in development mode"`
+
+	SnapTimeout int `long:"snaptimeout" description:"Unable to find a peer that supports snap-sync service for setting IBD timeout(seconds), it will switch to the normal sync method"`
```
Comment (dindinw) [inline config/config.go]: `SnapTimeout` -> `FindSnapSyncPeerTimeout`
a better description, `During the IBD phase, if no peer node with snap-sync service enabled is found after the timeout (seconds), the node will switch to using the normal sync method.`

**Review by dindinw — CHANGES_REQUESTED** [review]
Some textual revisions

File: config/config.go (hunk 2 above)
Comment (dindinw) [reply config/config.go]: or `NoSnapSyncPeerTimeout` is also OK for me

Comment (lochjin) [reply config/config.go]: https://github.com/Qitmeer/qng/pull/848/commits/a9bcdf4c8882bc4a69b6b50928a817cd7be46a2b

**Review by dindinw — APPROVED** [review]

**Pull request merged**